### PR TITLE
Split up and organize gdscript reference

### LINKED ISSUE #2735: Split up GDScript basics into many smaller sections.
https://docs.godotengine.org/en/3.1/getting_started/scripting/gdscript/gdscript_basics.html#doc-gdscript
This page is simply overwhelming. I never read it in full, only look up bits and pieces here or there when I need.
Splitting off exports (https://github.com/godotengine/godot-docs/pull/2718) is into one (or more) pages(s) is definitely a good idea, but can we have this for more of it's topics?

While it's called "Basics" of GDScript it actually seems like quite a comprehensive collection of it's features that go way beyond what I would consider basic. It's incredible valuable, overall very well explained information, but if was structured into smaller junks with pages (and more detailed list of the topics to left for each page), I would not get intimidated that easy, plus quickly get an overview about what information I can find here.

### PR DESCRIPTION
**Pull request by skyace65**
Splits up and organizes the gdscript reference page. Still working on it

- All information on creating custom classes is now on its own page in the gdscript section of the scripting documentation
- All information on creating signals has been moved to it's own page in the general scripting section. This page is going to need more work in the future to add C# and go over basic signal creation, right now it only has information on custom signals
- Removed the small History and export sections pointing to other pages. Also the table mentions the export page so there's still a link to it here
- The onready and tool annotation sections have been moved to after the annotations table
- The signal and assert keyword sections have been moved to after the keywords table

Possibly Closes #2735, still working on this, I'm going to see what else is reasonable to split into its own page, possibly the control flow information.

### CODE REVIEW

**Comment by pycbouh** [comment]
> * All information on creating custom classes is now on its own page in the gdscript section of the scripting documentation
> * All information on creating signals has been moved to it's own page in the general scripting section. This page is going to need more work in the future to add C# and go over basic signal creation, right now it only has information on custom signals

You shouldn't completely remove those sections from a page called "GDScript reference". This will make it much harder to look up information. Keep a short one or two paragraph covering how those concepts relate to GDScript and what GDScript-specific language features are there, then link to those new articles. This is especially important if you want to extend the articles to cover more than just GDScript.

Basically, a reference page has to keep all the information that is specific about the language, but general usage guides of the features available in the engine can be moved elsewhere.

-----
A big part that doesn't fit the "reference" style page is the section about "Built-in types". They are universal, and not GDScript specific. They can be moved elsewhere. Incidentally, this is now the only section that explains anything about signals, a concept that is quite unique to GDScript.
